community/modules/scheduler/pbspro-server/main.tf

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  resource_prefix = var.name_prefix != null ? var.name_prefix : "${var.deployment_name}-server"
+
+  user_startup_script_runners = var.startup_script == null ? [] : [
+    {
+      type        = "shell"
+      content     = var.startup_script
+      destination = "user_startup_script_pbs_server.sh"
+    }
+  ]
+}
+
+module "pbs_install" {
+  source = "github.com/GoogleCloudPlatform/hpc-toolkit//community/modules/scripts/pbspro-install?ref=1625555"
+
+  pbs_data_service_user   = var.pbs_data_service_user
+  pbs_exec                = var.pbs_exec
+  pbs_home                = var.pbs_home
+  pbs_license_server      = var.pbs_license_server
+  pbs_license_server_port = var.pbs_license_server_port
+
+  pbs_role = "server"
+  rpm_url  = var.pbs_server_rpm_url
+}
+
+module "pbs_qmgr" {
+  source = "github.com/GoogleCloudPlatform/hpc-toolkit//community/modules/scripts/pbspro-qmgr?ref=1625555"
+
+  client_host_count         = var.client_host_count
+  client_hostname_prefix    = var.client_hostname_prefix
+  execution_host_count      = var.execution_host_count
+  execution_hostname_prefix = var.execution_hostname_prefix
+  server_conf               = var.server_conf
+}
+
+module "server_startup_script" {
+  source = "github.com/GoogleCloudPlatform/hpc-toolkit//modules/scripts/startup-script?ref=64bc890"
```

Comment:
my thinking is that a reasonable facsimile of treating these as independent modules is to run `git log .` in the module directory and pick the commit you prefer. This has benefit of knowing reasonably easily whether you are using the latest and what the functionality differences are if not.

I don't think there is a `terraform init` benefit to using the same commits. It will blindly clone the same thing N times.

Reply:
ACK